`third-party/jemalloc/Makefile.target.include` not found when compiling on a HPE Cray EX system

This is a spinoff of #18758. Short summary below.

As per the title, when I try to compile Chapel programs on a HPE Cray EX system I get:
```
<prefix>/opt/chapel/make/Makefile.base:193: <prefix>/opt/chapel/third-party/jemalloc/Makefile.target.include: No such file or directory
gmake: *** No rule to make target '<prefix>/opt/chapel/third-party/jemalloc/Makefile.target.include'.  Stop.
error: compiling generated source
```

Here's some diagnostic info:
```
$ echo $CHPL_HOME
<prefix>/opt/chapel

$ chpl --savec build hello.chpl
<prefix>/opt/chapel/make/Makefile.base:193: <prefix>/opt/chapel/third-party/jemalloc/Makefile.target.include: No such file or directory
gmake: *** No rule to make target '<prefix>/opt/chapel/third-party/jemalloc/Makefile.target.include'.  Stop.
error: compiling generated source

$ grep '^CHPL' build/Makefile
CHPL_MAKE_HOME = <prefix>/opt/chapel
CHPL_MAKE_RUNTIME_LIB = <prefix>/opt/chapel/lib
CHPL_MAKE_RUNTIME_INCL = <prefix>/opt/chapel/runtime/include
CHPL_MAKE_THIRD_PARTY = <prefix>/opt/chapel/third-party
CHPLSRC = \
CHPLUSEROBJ = \

$ $CHPL_HOME/util/printchplenv --anonymize --all
CHPL_HOST_PLATFORM: hpe-cray-ex
CHPL_HOST_COMPILER: gnu
  CHPL_HOST_CC: gcc
  CHPL_HOST_CXX: g++
CHPL_HOST_ARCH: x86_64
CHPL_TARGET_PLATFORM: hpe-cray-ex
CHPL_TARGET_COMPILER: cray-prgenv-cray
  CHPL_TARGET_CC: cc
  CHPL_TARGET_CXX: CC
CHPL_TARGET_ARCH: x86_64
CHPL_TARGET_CPU: x86-rome
CHPL_LOCALE_MODEL: flat
CHPL_COMM: ofi
  CHPL_LIBFABRIC: system
CHPL_TASKS: qthreads
CHPL_LAUNCHER: slurm-srun
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_HOST_MEM: cstdlib
CHPL_MEM: jemalloc
CHPL_ATOMICS: cstdlib
  CHPL_NETWORK_ATOMICS: ofi
CHPL_GMP: bundled
CHPL_HWLOC: bundled
CHPL_RE2: bundled
CHPL_LLVM: none +
CHPL_AUX_FILESYS: none
CHPL_LIB_PIC: none
CHPL_SANITIZE: none
CHPL_SANITIZE_EXE: none

$ module list

Currently Loaded Modules:
  1) cce/11.0.4     3) craype-x86-rome         5) craype-network-ofi       7) xpmem/2.2.40-7.0.1.0_2.7__g1d7a24d.shasta   9) cray-libsci/21.04.1.1  11) bolt/0.7        13) load-epcc-module  15) cray-pmi-lib/6.0.10
  2) craype/2.7.6   4) libfabric/1.11.0.4.71   6) perftools-base/21.02.0   8) cray-mpich/8.1.4                           10) PrgEnv-cray/8.0.0      12) epcc-setup-env  14) cray-pmi/6.0.10
```

Here's the program I was trying to compile:
```chapel
$ cat hello.chpl 
use Memory.Diagnostics;


serial do coforall loc in Locales do on loc {
  writeln("locale #", here.id, "...");
  writeln("  ...is named: ", here.name);
  writeln("  ...has hostname: ", here.hostname);
  writeln("  ...has ", here.numPUs(), " processor cores");
  writeln("  ...has ", here.physicalMemory(unit=MemUnits.GB, retType=real),
          " GB of memory");
  writeln("  ...has ", here.maxTaskPar, " maximum parallelism");
}

writeln("Total number of cores: ", + reduce Locales.numPUs());
writeln("Total memory: ", + reduce Locales.physicalMemory(unit=MemUnits.GB, retType=real), " GB");
writeln("Total maximum parallelism: ", + reduce Locales.maxTaskPar);
```

Tagging @aconsroe-hpe.

--- a/util/buildRelease/install.sh
+++ b/util/buildRelease/install.sh
@@ -303,13 +303,13 @@ cd ..
 for dir in $THIRD_PARTY_DIRS
 do
   #echo "Considering 3p dir $dir"
-  if [ -f third-party/"$dir"/Makefile.include ]
-  then
-    for f in third-party/"$dir"/Makefile*
-    do
+  for f in third-party/"$dir"/Makefile*
+  do
+    if [ -f "$f" ]
+    then
       myinstallfile "$f"  "$DEST_THIRD_PARTY"/"$dir"
-    done
-  fi
+    fi
+  done
   if [ -d third-party/"$dir"/install ]
   then
     myinstalldir "third-party/$dir/install" "$DEST_THIRD_PARTY/$dir/install/"

